mhs/common/mhs_common/workflow/common.py

```diff
@@ -63,9 +72,17 @@ def __init__(self, routing: routing_reliability.RoutingAndReliability = None):
             url = CommonWorkflow._extract_endpoint_url(endpoint_details)
             to_party_key = endpoint_details[MHS_TO_PARTY_KEY_KEY]
             cpa_id = endpoint_details[MHS_CPA_ID_KEY]
-            logger.info('0002', 'Retrieved endpoint details for {service_id}. {url}, {to_party_key}, {cpa_id}',
-                        {'service_id': service_id, 'url': url, 'to_party_key': to_party_key, 'cpa_id': cpa_id})
-            return url, to_party_key, cpa_id
+            to_asid = self._extract_asid(endpoint_details)
+            details = {self.ENDPOINT_SERVICE_ID: service_id,
+                       self.ENDPOINT_URL: url,
+                       self.ENDPOINT_PARTY_KEY: to_party_key,
+                       self.ENDPOINT_CPA_ID: cpa_id,
+                       self.ENDPOINT_TO_ASID: to_asid
+                       }
+            logger.info('0002',
+                        'Retrieved endpoint details for {service_id}. {url}, {party_key}, {cpa_id} {to_asid}',
```

Comment:
Not something you have to change now, but I think the logger does pretty well at logging dictionary contents, so it might mean fewer future changes to do:
```suggestion
                        'Retrieved endpoint details for {service_id}. {details}, details)',
```